cclient/iota_client_extended_api.c

```diff
@@ -749,17 +765,55 @@ retcode_t iota_client_promote_transaction(
 
   if (consistency_res->state) {
     // adding spam transaction
-    spam_transaction = transaction_new();
-    if (!spam_transaction) {
+    transaction_reset(&spam_transaction);
```

Comment:
we should free the consistency req/res now because we no longer need their content

Reply:
done :)